applications/ChimeraApplication/custom_strategies/strategies/fs_strategy_for_chimera.h

```diff
@@ -0,0 +1,836 @@
+
+                    if(NodalArea >mAreaTolerance)
+                    {
+                        if ( ! itNode->IsFixed(VELOCITY_X) )
+                            itNode->FastGetSolutionStepValue(VELOCITY_X) += itNode->FastGetSolutionStepValue(FRACT_VEL_X) / NodalArea;
+                        if ( ! itNode->IsFixed(VELOCITY_Y) )
+                            itNode->FastGetSolutionStepValue(VELOCITY_Y) += itNode->FastGetSolutionStepValue(FRACT_VEL_Y) / NodalArea;
+                    } else {
+                        KRATOS_WARNING_ONCE("Nodal area too small ! ")<<std::endl;
+                    }
+                }
+            }
+
+        auto &pre_modelpart = rModelPart.GetSubModelPart("fs_pressure_model_part");
+        const auto& r_constraints_container = pre_modelpart.MasterSlaveConstraints();
+            for(const auto& constraint : r_constraints_container)
+            {
+                const auto& slave_dofs = constraint.GetSlaveDofsVector();
+                for(const auto& slave_dof : slave_dofs)
+                {
+                    const auto slave_node_id = slave_dof->Id(); // DOF ID is same as node ID
+                    auto& r_slave_node = rModelPart.Nodes()[slave_node_id];
+                    r_slave_node.FastGetSolutionStepValue(VELOCITY_X)=0;
+                    r_slave_node.FastGetSolutionStepValue(VELOCITY_Y)=0;
+                }
+            }
+
+            for(const auto& constraint : r_constraints_container)
+            {
+                const auto& master_dofs = constraint.GetMasterDofsVector();
+                const auto& slave_dofs = constraint.GetSlaveDofsVector();
+                ModelPart::MatrixType r_relation_matrix;
+                ModelPart::VectorType r_constant_vector;
+                constraint.CalculateLocalSystem(r_relation_matrix,r_constant_vector,rModelPart.GetProcessInfo());
+
+                IndexType slave_i = 0;
+                for(const auto& slave_dof : slave_dofs)
+                {
+                    const auto slave_node_id = slave_dof->Id(); // DOF ID is same as node ID
+                    auto& r_slave_node = rModelPart.Nodes()[slave_node_id];
+                    IndexType master_j = 0;
+                    for(const auto& master_dof : master_dofs)
+                    {
+                        const auto master_node_id = master_dof->Id();
+                        const double weight = r_relation_matrix(slave_i, master_j);
+                        auto& r_master_node = rModelPart.Nodes()[master_node_id];
+
+                        r_slave_node.FastGetSolutionStepValue(VELOCITY_X) +=(r_master_node.FastGetSolutionStepValue(VELOCITY_X))*weight;
+                        r_slave_node.FastGetSolutionStepValue(VELOCITY_Y) +=(r_master_node.FastGetSolutionStepValue(VELOCITY_Y))*weight;
+
+                        ++master_j;
+                    }
+                    ++slave_i;
+                }
+            }
+        }
```

Comment:
I'd split this in 2 functions, one for the 2D case and one for the 3D one (it would make the file much more readable).

Reply:
Done !